### PR TITLE
Add TRUNCATE arbitrary config tests

### PR DESCRIPTION
**Pull request by gledis69**
* Also adds the ability to skip tests from particular configs.

### CODE REVIEW

**Comment by gledis69** [comment]
Currently the arbitrary config tests fail due to this #5853.

File: src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py [hunk 1]
```diff
@@ -128,13 +128,26 @@ def copy_test_files(config):
...
 
 
-def copy_test_files_with_names(test_names, sql_dir_path, expected_dir_path):
+def copy_test_files_with_names(test_names, sql_dir_path, expected_dir_path, config):
     for test_name in test_names:
+        # make empty files for the skipped tests
+        if test_name in config.skip_tests:
+            expected_sql_file = os.path.join(sql_dir_path, test_name + ".sql")
+            with open(expected_sql_file, 'w') as f:
+                f.write("")
```
Comment (agedemenli) [inline src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py:140]: Do we need the `write` function here? We actually don't write anything.
Not sure but `open` with mode "a" could simply do what we need

**Review by agedemenli — COMMENTED** [review]
Thank you for extensive testing. How about adding a simple test to truncate foreign tables?

**Comment by agedemenli** [comment]
> How about adding a simple test to truncate foreign tables?

We can also test this in a separate PR, while testing foreign tables in arbitrary configs.
I'm ok with both options

File: src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py (hunk 1 above)
Comment (gledis69) [reply src/test/regress/citus_tests/arbitrary_configs/citus_arbitrary_configs.py:140]: I can actually probably reduce these likes to `open(expected_sql_file, 'x').close()`.

**Review by agedemenli — APPROVED** [review]
LGTM.
Enterprise conflict is probably something really simple, might be in the create_schedule order.
Please don't forget to squash before merging.
Adding one more reference table that is connected to the main fkey graph would be nice, but I'm ok merging without it as well.
Thank you

**Pull request merged**